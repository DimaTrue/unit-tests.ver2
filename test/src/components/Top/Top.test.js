import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Top from './Top';

Enzyme.configure({ adapter: new Adapter() });

describe ('renders', () => {

it('render TOP component w/o props', () => {
    const component = shallow(<Top/>);
    expect(component.find('ul')).toHaveLength(1);
    expect(component.find('li')).toHaveLength(0);
}); 

it('render TOP component with props', () => {
    const component = shallow(<Top list={['item1', 'item2', 'item4', 'item5']}/>);
    expect(component.find('ul')).toHaveLength(1);
    expect(component.find('li')).toHaveLength(4);
}) 

it('match snapshot', () => {
    const component = shallow(<Top list={[1, 2, 3, 4]}/>);
    expect(component).toMatchSnapshot();
})

});