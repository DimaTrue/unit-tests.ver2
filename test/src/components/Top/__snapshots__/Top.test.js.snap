exports[`renders match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Top
    list={
        Array [
            1,
            2,
            3,
            4,
          ]
    } />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li>
          1
</li>,
        <li>
          2
</li>,
        <li>
          3
</li>,
        <li>
          4
</li>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": 1,
        },
        "ref": null,
        "rendered": 1,
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": 2,
        },
        "ref": null,
        "rendered": 2,
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": 3,
        },
        "ref": null,
        "rendered": 3,
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "children": 4,
        },
        "ref": null,
        "rendered": 4,
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            1
</li>,
          <li>
            2
</li>,
          <li>
            3
</li>,
          <li>
            4
</li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": 1,
          },
          "ref": null,
          "rendered": 1,
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": 2,
          },
          "ref": null,
          "rendered": 2,
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": 3,
          },
          "ref": null,
          "rendered": 3,
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": 4,
          },
          "ref": null,
          "rendered": 4,
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
